app/Http/Controllers/MaterialController.php

```diff
@@ -89,7 +91,42 @@ public function edit($id)
      */
     public function update(Request $request, $id)
     {
-        //
+        //TODO: Refactor: Look into using sync to avoid deleting entries every time we update.
+
+        try {
+            $request->validate([
+                'assembly_id' => 'required|integer|min:1',
```

Comment:
Ah spotted a problem, since we're no longer using `$request->assembly_id`, having this here will cause the request to fail.